.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimeExtractor.cs

```diff
@@ -163,5 +167,26 @@ private List<Token> SpecialCasesRegexMatch(string text, DateObject reference)
 
             return result;
         }
+
+        private List<ExtractResult> FilterAmbiguity(List<ExtractResult> extractResults, string text)
+        {
+            if (this.config.AmbiguityFiltersDict != null)
+            {
+                foreach (var regex in this.config.AmbiguityFiltersDict)
+                {
+                    foreach (var extractResult in extractResults)
+                    {
+                        if (regex.Key.IsMatch(extractResult.Text))
+                        {
+                            var matches = regex.Value.Matches(text).Cast<Match>();
+                            extractResults = extractResults.Where(er => !matches.Any(m => m.Index + m.Length == er.Start))
```

Comment:
The condition here is too dependent on the specific regex. People changing the resources file will not be aware of the criteria here. Ideally it should be only a check if a match is within the other.

Reply:
Agreed, however here it is crucial to check that the match and the extractResult are concatenated. One could change the second regex in AmbiguityTimeFiltersDict to '(pour|durée\s+de)\s+(.+)' and then check if its match contains the extractResult, but then cases like "Je vais partir pour la mer à 3 heures" (I will leave for the sea at 3 o'clock) will also be treated as duration.